Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -2050,6 +2085,119 @@ def pin_message():
+    conversations = integration_context.get('conversations')
+    if conversations:
+        conversations = json.loads(conversations)
+        conversations.extend(updated_channel_list)
+    else:
+        conversations = updated_channel_list
+    set_to_integration_context_with_retries({'conversations': conversations}, OBJECTS_TO_KEYS, SYNC_CONTEXT)
+
+
+def fetch_channels_from_api(conversation_name: str = ''):
+    """
+    Fetches channel IDs based on their name, if given a conversation name, it will search for a specific ID
+
+    Args:
+        conversation_name: Optional - The human readable form of a channel's name.
+
+    Returns:
+        A list of dictionaries containing a channel name and ID, or a singular name and ID if given a conversation name.
+    """
+    demisto.debug('Fetching channels')
+    cursor = None
+    updated_channel_list: list = []
+
+    res = CLIENT.conversations_list(types='private_channel,public_channel', limit=PAGINATED_COUNT,
+                                    exclude_archived=True, cursor=cursor)
+
+    while True:
+        cursor = res.get('response_metadata', {}).get('next_cursor')
+        channels = res['channels'] if res and res.get('channels') else []
+        if conversation_name:
+            conversation_filter = list(filter(lambda c: c.get('name') == conversation_name, channels))
+            if conversation_filter:
+                return conversation_filter[0]
+        else:
+            for channel in channels:
+                updated_channel: dict = {
+                    'name': channel.get('name'),
+                    'id': channel.get('id')
+                }
+                updated_channel_list.append(updated_channel)
+        if not cursor:
+            demisto.info("Reached the end of looking for a channel")
+            break
+        else:
+            total_try_time = 0
+            while True:
+                try:
+                    res = CLIENT.conversations_list(types='private_channel,public_channel', exclude_archived=True,
+                                                    cursor=cursor)
+                except SlackApiError as api_error:
+                    demisto.debug(f'Got rate limit error (sync). Body is: {str(res)}\n{api_error}')
+                    response = api_error.response
+                    headers = response.headers  # type: ignore
+                    if 'Retry-After' in headers:
+                        retry_after = int(headers['Retry-After'])
+                        total_try_time += retry_after
+                        if total_try_time < MAX_LIMIT_TIME:
+                            time.sleep(retry_after)
+                            continue
+                    else:
+                        if total_try_time < MAX_LIMIT_TIME:
+                            time.sleep(5)
+                            continue
+                        else:
+                            raise
+                break
+    return updated_channel_list
```

Comment:
I think that if this function gets a conversation name that does not exist it will crush as `updated_channel_list` does not exist. please add a unit test for this first. (has conversation name and reaching the end of the cursor.

Reply:
So it didn't crash, but it did return an empty list. I fixed the code to return an error instead. We had a unit test which covered this, but the assertion was that the returned conversation was `None`, but an empty list also returns as a `None` type.

Covered in this test - test_get_conversation_by_name_not_exists